core/api/src/services/dealer-quote/dealer-quote.ts

```diff
@@ -0,0 +1,202 @@
+
+const clientGetQuoteToSellUsd = util.promisify<
+  GetQuoteToSellUsdRequest,
+  GetQuoteToSellUsdResponse
+>(client.getQuoteToSellUsd.bind(client))
+
+const clientAcceptQuote = util.promisify<AcceptQuoteRequest, AcceptQuoteResponse>(
+  client.acceptQuote.bind(client),
+)
+
+export const DealerQuoteService = (): IDealerQuoteService => {
+  const getQuoteToBuyUsdWithSats = async function (
+    btcAmount: BtcPaymentAmount,
+    immediateExecution: boolean = false,
+  ): Promise<QuoteToBuyUsd | DealerQuoteServiceError> {
+    try {
+      const req = new GetQuoteToBuyUsdRequest().setAmountToSellInSats(
+        Number(btcAmount.amount),
+      )
+      req.setImmediateExecution(immediateExecution)
+
+      console.log("REQ: ", req)
+
+      const response = await clientGetQuoteToBuyUsd(req)
+
+      console.log("response: ", response)
+
+      const quote = response.toObject() as QuoteToBuyUsd
+
+      return quote
+    } catch (error) {
+      baseLogger.error({ error }, "GetQuoteToBuyUsdWithSats unable to fetch quote")
+      return handleDealerErrors(error)
+    }
+  }
+
+  const getQuoteToBuyUsdWithCents = async function (
+    usdAmount: UsdPaymentAmount,
+    immediateExecution = false,
+  ): Promise<QuoteToBuyUsd | DealerQuoteServiceError> {
+    try {
+      const req = new GetQuoteToBuyUsdRequest().setAmountToBuyInCents(
+        Number(usdAmount.amount),
+      )
+      req.setImmediateExecution(immediateExecution)
+
+      console.log("REQ: ", req)
+
+      const response = await clientGetQuoteToBuyUsd(req)
+
+      console.log("response: ", response)
+
+      const quote = response.toObject() as QuoteToBuyUsd
+
+      return quote
+    } catch (error) {
+      baseLogger.error({ error }, "GetQuoteToBuyUsdWithCents unable to fetch quote")
+      return handleDealerErrors(error)
+    }
+  }
+
+  const getQuoteToSellUsdWithSats = async function (
+    btcAmount: BtcPaymentAmount,
+    immediateExecution = false,
+  ): Promise<QuoteToSellUsd | DealerQuoteServiceError> {
+    try {
+      const req = new GetQuoteToSellUsdRequest().setAmountToBuyInSats(
+        Number(btcAmount.amount),
+      )
+      req.setImmediateExecution(immediateExecution)
+
+      const response = await clientGetQuoteToSellUsd(req)
+
+      const quote = response.toObject() as QuoteToSellUsd
+
+      return quote
+    } catch (error) {
+      baseLogger.error({ error }, "GetQuoteToSellUsdWithSats unable to fetch quote")
+      return handleDealerErrors(error)
+    }
+  }
+
+  const getQuoteToSellUsdWithCents = async function (
+    usdAmount: UsdPaymentAmount,
+    immediateExecution = false,
+  ): Promise<QuoteToSellUsd | DealerQuoteServiceError> {
+    try {
+      const req = new GetQuoteToSellUsdRequest().setAmountToSellInCents(
+        Number(usdAmount.amount),
+      )
+      req.setImmediateExecution(immediateExecution)
+
+      const response = await clientGetQuoteToSellUsd(req)
+
+      const quote = response.toObject() as QuoteToSellUsd
```

Comment:
create the proper translate method (check other services as reference). Apply to all the methods in the service